charts/openstack-cloud-controller-manager/templates/daemonset.yaml

```diff
@@ -25,6 +25,10 @@ spec:
       labels:
         {{- include "occm.controllermanager.labels" . | nindent 8 }}
     spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
```

Comment:
I am not entirely sure this works with the default of `imagePullSecrets: []` that is currently being set in the values file.

We have to make sure that this works in the following cases:

1. Default case (user does not set `imagePullSecrets` and they are `imagePullSecrets: []`
2. Single element in `imagePullSecrets`
3. Multiple elements in `imagePullSecrets`

This might require a combination of `if`, `range` and/or `with` to achive the desired outcome.

Reply:
With the values file, the output should be empty, but once we overwrite the values.yml with
```
imagePullSecrets:
- one
```
The output will be:
```
imagePullSecrets:
        - one
```
It also works with more values, it can be tested with 
`helm template --values values.yaml .`

I updated the last commit with a few comments.

Reply:
lgtm, tested with helm template and all possibilities render correctly.